pkg/operation/botanist/controlplane/clusterautoscaler/cluster_autoscaler.go

```diff
@@ -0,0 +1,497 @@
+		codec    = kubernetes.ShootCodec.CodecForVersions(kubernetes.ShootSerializer, kubernetes.ShootSerializer, versions, versions)
+
+		clusterRole = &rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "system:cluster-autoscaler-shoot",
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{""},
+					Resources: []string{"events", "endpoints"},
+					Verbs:     []string{"create", "patch"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"pods/eviction", "configmaps"},
+					Verbs:     []string{"create"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"pods/status"},
+					Verbs:     []string{"update"},
+				},
+				{
+					APIGroups:     []string{""},
+					Resources:     []string{"endpoints"},
+					ResourceNames: []string{ServiceName},
+					Verbs:         []string{"get", "update"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"nodes"},
+					Verbs:     []string{"watch", "list", "get", "update"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"pods", "services", "replicationcontrollers", "persistentvolumeclaims", "persistentvolumes"},
+					Verbs:     []string{"watch", "list", "get"},
+				},
+				{
+					APIGroups: []string{"apps", "extensions"},
+					Resources: []string{"daemonsets", "replicasets", "statefulsets"},
+					Verbs:     []string{"watch", "list", "get"},
+				},
+				{
+					APIGroups: []string{"policy"},
+					Resources: []string{"poddisruptionbudgets"},
+					Verbs:     []string{"watch", "list"},
+				},
+				{
+					APIGroups: []string{"storage.k8s.io"},
+					Resources: []string{"storageclasses"},
+					Verbs:     []string{"watch", "list", "get"},
+				},
+				{
+					APIGroups:     []string{""},
+					Resources:     []string{"configmaps"},
+					ResourceNames: []string{"cluster-autoscaler-status"},
+					Verbs:         []string{"delete", "get", "update"},
+				},
+			},
+		}
+		clusterRoleYAML, _ = runtime.Encode(codec, clusterRole)
+
+		clusterRoleBinding = &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "system:cluster-autoscaler-shoot",
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "ClusterRole",
+				Name:     clusterRole.Name,
+			},
+			Subjects: []rbacv1.Subject{{
+				Kind: rbacv1.UserKind,
+				Name: UserName,
+			}},
+		}
+		clusterRoleBindingYAML, _ = runtime.Encode(codec, clusterRoleBinding)
+	)
+
+	return map[string][]byte{
+		"clusterrole.yaml":        clusterRoleYAML,
+		"clusterrolebinding.yaml": clusterRoleBindingYAML,
+	}
+}
+
+// Secrets is collection of secrets for the cluster-autoscaler.
+type Secrets struct {
+	// Kubeconfig is a secret which can be used by the cluster-autoscaler to communicate to the kube-apiserver.
+	Kubeconfig component.Secret
+}
+
+// BootstrapSeed deploys the RBAC configuration for the control cluster.
+func BootstrapSeed(ctx context.Context, c client.Client, namespace, _ string) error {
```

Comment:
Maybe it would make sense to put this function into its own file e.g `cluster_autoscaler_seed_bootstrap.go`?.
I just did not realise at first that this is only used during the Seed bootstrap, while the other contents for the file are used during the Shoot reconciliation.

Reply:
Yep, makes sense, will do! Thanks.